File: chat-web/types/events/event.ts
import { EventType } from "./eventtype";
import { MessageCreated } from "./messagecreated";
import { MessageUpdated } from "./messageupdated";

export type Event =
  { t: EventType.MESSAGE_CREATED; d: MessageCreated }
  | { t: EventType.MESSAGE_UPDATED; d: MessageUpdated }
;

File: chat-api/chat_types/events/event.py
from .event_type import EventType
from .channel_created import ChannelCreated
from .message_created import MessageCreated
from .message_updated import MessageUpdated


Event = {"t": EventType.CHANNEL_CREATED, "d": ChannelCreated} | {"t": EventType.MESSAGE_CREATED, "d": MessageCreated} | {"t": EventType.MESSAGE_UPDATED, "d": MessageUpdated}
